### ISSUE TITLE
SC2126: False warning

### ISSUE THREAD

**Issue by Akianonymus** [koalaman/shellcheck#1920]
- My shellcheck version: Online
- The rule's [wiki page](https://shellcheck.net/wiki/SC2126) does not already cover this.

#### Here's what shellcheck currently says:

```
    grep 'saved\|retrieved' ./*/wget*log* | wc -l
    ^-- SC2126: Consider using grep -c instead of grep|wc -l.

```
Currently, I want to grep multiple files for the matching string and print the matching line, then count it.
Using suggested solution gives individual file count.

```
grep -c 'saved\|retrieved' ./*/wget*log*
    ./wget-log:0
    ./wget-log.1:1
    ./wget-log:2:1
    ./wget-log.3:1

```

Using my command.
```
grep 'saved\|retrieved' ./*/wget*log* | wc -l
    3
```

**Comment by brother** [comment]
When considered the warning put a ignore directive in and continue on. That's how I treat these messages, it's hard to know what your intentions are.

**Comment by mnuccioarpae** [comment]
You can avoid running `wc` and use only `grep` and some bash magics, but it's a bit cryptic:

`echo $(($(c=($(grep -hc 'saved\|retrieved' ./*/wget*log*)); IFS=+; echo "${c[*]}")))`

**Comment by Akianonymus** [comment]
@brother Yeah, i did that.

Also, ig it's not that hard to know my intentions as this is only a problem for globs, ig a check could be added in source.

Again, just a thought, this isn't something important.

**Comment by Akianonymus** [comment]
@mnuccioarpae Well, bash magics are good, cryptic or not.

Thank you for the help.

**Comment by dimo414** [comment]
No need for all that "bash magic", just pipe the contents into `grep`:

    cat ./*/wget*log* | grep -c 'saved\|retrieved'
